### PR TITLE
T4 Support

### PR DESCRIPTION
**Pull request by KurtE**
I have not tested all pins yet, but I have done simple test of T4 through prop shield to 8 neopixel bar and it worked on all of those valid Serial TX pins...

Updated examples with comments likewise with readme.

Warning: Requires fix in core library as the file teensy4\imxrt.h has the DMAMUX_SOURCE values wrong (TX and RX pins were reversed)

### CODE REVIEW

**Comment by PaulStoffregen** [comment]
Thanks!

File: examples/BasicTest/BasicTest.ino [hunk 1]
```diff
@@ -14,6 +14,7 @@ const int pin = 1;
 //   Teensy 3.2:  1, 5, 8, 10, 31   (overclock to 120 MHz for pin 8)
 //   Teensy 3.5:  1, 5, 8, 10, 26, 32, 33, 48
 //   Teensy 3.6:  1, 5, 8, 10, 26, 32, 33
+//   Teensy 4.0:  1, 8, 14, 16, 20, 24, 29, 39
```
Comment (rjongeneelen) [inline examples/BasicTest/BasicTest.ino:17]: Shouldn't 16 be 17 (TX4)?

**Review by rjongeneelen — COMMENTED** [review]
Wrong pin?

**Comment by KurtE** [comment]
Yep probably wrong pin in comment…

**Pull request merged**